### DISCUSSION TITLE
BNF expression parser

### DISCUSSION THREAD

**Discussion by ZLangJIT** [foonathan/lexy#212] [Q&A]
is there any work on making a constexpr BNF expression parser for easier use of lexy ?

as lexy is very powerful but very complicated due to how compilers emit template errors and complicated rule/branch matching

**Comment by foonathan** [comment]
That is not planned. lexy is by design not declarative, but instead syntax sugar for a recursive descent parser. BNF is a declarative way that automatically figures out the rule/branch matching to some algorithm with unclear performance characteristics and power. If you want that, you should look into parser generators.
